tests/EmbeddedRelationsTest.php

```diff
@@ -972,4 +974,18 @@ public function testUnsetPropertyOnEmbed()
         $this->assertNull($user->addresses->get(0)->city);
         $this->assertSame('Kyoto', $user->addresses->get(1)->city);
     }
+
+    public function testEmbedManyToArrayCast()
+    {
+        $user = UserWithEmbeds::create(['name' => 'John Doe']);
+        $user->addresses()->saveMany([new Address(['city' => 'London'])]);
+
+        //Reload document
+        $user = UserWithEmbeds::where('name', 'John Doe')->first();
```

Comment:
In order to get the `address` property correctly casted, you must call `with('address')`:
```suggestion
        $user = UserWithEmbeds::where('name', 'John Doe')->with('addresses')->first();
```